### ISSUE TITLE
`extra` field in `registered_meta` is serialized both as an object and as an array, making deserialization difficult 

### ISSUE THREAD

**Issue by emetsger** [DataConservancy/dcs-packaging-osf#2]
The `registered_meta` field carries an object named `summary`.  The `summary` object carries a field named `extra`.  

I have two registrations; in one `registered_meta->summary->extra` is an empty object, and in another it is an empty array:
* https://api.osf.io/v2/registrations/tgzhk/?version=2.2
* https://api.osf.io/v2/registrations/5nua2/?version=2.2

`extra` is an array:
```json  
  "registered_meta": {
    "summary": {
      "value": "test",
      "comments": [],
      "extra": []
    }
  }
```

`extra` is an object:
```json
"registered_meta": {
    "summary": {
      "value": "Test registration.",
      "comments": [],
      "extra": {}
    }
  }
```

**Comment by birkland** [comment]
Is this a bug in the osf API, or is it intended to be this way?

**Comment by emetsger** [comment]
Opened https://github.com/CenterForOpenScience/osf.io/issues/6696 to track this on the OSF side.

**Comment by emetsger** [comment]
COS confirmed that the `extra` type in this case should be an object, not an array.

**Comment by msuhovec** [comment]
Following up from a discussion we had a few weeks back:

Looking at https://github.com/CenterForOpenScience/osf.io/issues/6696

I see a comment from 3/17 saying this has been fixed. Does the tool need to be repackaged to test this? Is this still the active repo for the command line  client?

**Comment by emetsger** [comment]
Hi @msuhovec!  You can try the latest client, version 1.1.1 found at http://maven.dataconservancy.org/public/releases/org/dataconservancy/cos/osf-packager-cli/1.1.1/osf-packager-cli-1.1.1.jar

Let me know if you have problems!!

**Comment by msuhovec** [comment]
@emetsger 

curatewkrpprd.library.nd.edu: java -version
java version "1.8.0_101"
Java(TM) SE Runtime Environment (build 1.8.0_101-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.101-b13, mixed mode)

I'm still getting the data serialization error:

 java -jar ./osf-packager-cli-1.1.1.jar -c /opt/batchs/tasks/osf-cli.conf -n 3gvg9 -o 3gvg9 https://api.osf.io/v2/registrations/3gvg9/
Destination directory /global/home/msuhovec/3gvg9 already exists!  Either (re)move the directory, or choose a different output location.
curatewkrpprd.library.nd.edu: rmdir 3gvg9
curatewkrpprd.library.nd.edu: java -jar ./osf-packager-cli-1.1.1.jar -c /opt/batchs/tasks/osf-cli.conf -n 3gvg9 -o 3gvg9 https://api.osf.io/v2/registrations/3gvg9/
There was an unrecoverable error: com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of java.util.LinkedHashMap out of START_ARRAY token
 at [Source: N/A; line: -1, column: -1] (through reference chain: org.dataconservancy.cos.osf.client.model.Registration["registered_meta"]->java.util.LinkedHashMap["item32"]->org.dataconservancy.cos.osf.client.model.RegistrationMetadata["extra"])
java.lang.RuntimeException: com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of java.util.LinkedHashMap out of START_ARRAY token
 at [Source: N/A; line: -1, column: -1] (through reference chain: org.dataconservancy.cos.osf.client.model.Registration["registered_meta"]->java.util.LinkedHashMap["item32"]->org.dataconservancy.cos.osf.client.model.RegistrationMetadata["extra"])
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectInternal(ResourceConverter.java:224)
	at com.github.jasminb.jsonapi.ResourceConverter.readObject(ResourceConverter.java:184)
	at org.dataconservancy.cos.osf.client.retrofit.PaginatedResponseBodyConverter.convert(PaginatedResponseBodyConverter.java:84)
	at org.dataconservancy.cos.osf.client.retrofit.PaginatedResponseBodyConverter.convert(PaginatedResponseBodyConverter.java:31)
	at retrofit.OkHttpCall.parseResponse(OkHttpCall.java:148)
	at retrofit.OkHttpCall.execute(OkHttpCall.java:116)
	at org.dataconservancy.cos.packaging.cli.PackageGenerationApp.run(PackageGenerationApp.java:218)
	at org.dataconservancy.cos.packaging.cli.PackageGenerationApp.main(PackageGenerationApp.java:190)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: Can not deserialize instance of java.util.LinkedHashMap out of START_ARRAY token
 at [Source: N/A; line: -1, column: -1] (through reference chain: org.dataconservancy.cos.osf.client.model.Registration["registered_meta"]->java.util.LinkedHashMap["item32"]->org.dataconservancy.cos.osf.client.model.RegistrationMetadata["extra"])
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:216)
	at com.fasterxml.jackson.databind.DeserializationContext.mappingException(DeserializationContext.java:873)
	at com.fasterxml.jackson.databind.DeserializationContext.mappingException(DeserializationContext.java:869)
	at com.fasterxml.jackson.databind.deser.std.StdDeserializer._deserializeFromEmpty(StdDeserializer.java:874)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:337)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:26)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:490)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:95)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:260)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:125)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer._readAndBindStringMap(MapDeserializer.java:495)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:341)
	at com.fasterxml.jackson.databind.deser.std.MapDeserializer.deserialize(MapDeserializer.java:26)
	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:490)
	at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:95)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:260)
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:125)
	at com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:3760)
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2042)
	at com.fasterxml.jackson.databind.ObjectMapper.treeToValue(ObjectMapper.java:2529)
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectInternal(ResourceConverter.java:304)
	at com.github.jasminb.jsonapi.ResourceConverter.readObjectInternal(ResourceConverter.java:208)
	... 7 more

**Comment by emetsger** [comment]
@msuhovec that may be a different error, I can take a look!

**Comment by emetsger** [comment]
@msuhovec this error stems from the fact that the Registration [form](https://osf.io/3gvg9/register/564d31db8c5e4a7c9694b2c4) for [registration 3gvg9](https://osf.io/3gvg9/) carries a bunch of information that the packaging tool isn't equipped to handle at the moment.  This is an issue distinct from https://github.com/CenterForOpenScience/osf.io/issues/6696.  

I'll open a new issue for this problem!

**Comment by msuhovec** [comment]
So, for my own information, what fields are verboten?

**Comment by emetsger** [comment]
@msuhovec IIRC when you create a registration using the OSF, the user is presented with a choice of the type of registration to make:

<img width="885" alt="screen shot 2017-05-19 at 2 55 20 pm" src="https://cloud.githubusercontent.com/assets/146970/26262581/443fa8c4-3ca3-11e7-826f-178e4e3eaa32.png">

Depending on the selection, the user is presented with different form fields, inputting information relating to the registration.

The "Open-Ended" registration is the one supported by the packaging tool; it simply requires a summary:

<img width="951" alt="screen shot 2017-05-19 at 2 56 57 pm" src="https://cloud.githubusercontent.com/assets/146970/26262662/8921d034-3ca3-11e7-9bc3-33faab2826f3.png">

Other registration types will ask for different information, and the packaging tool doesn't know how to support them.

**Comment by msuhovec** [comment]
Elliot-

I just created an open-ended registration for that project, and I get the
same results.

Do you have a project registration that works for you that I could try?

**Comment by emetsger** [comment]
@msuhovec yikes!  Ok, I will investigate!

**Comment by emetsger** [comment]
@msuhovec a quick update:

So there is a bug to be fixed locally, and I have a fix waiting in the wings.  While attempting to confirm the fix works, I ran into some other problems that seem to be on the OSF's side.  I'm working with them to see what's up.  I just wanted to let you know a release will be made as soon as I can confirm the fix!

**Comment by msuhovec** [comment]
Thanks Eliiot- I really appreciate you investigating this!

**Comment by emetsger** [comment]
Addressed by #34, confirmed by @msuhovec
